kernel/src/expressions/scalars.rs

```diff
@@ -20,8 +20,8 @@ pub struct ArrayData {
 }
 
 impl ArrayData {
-    #[cfg(test)]
-    pub(crate) fn new(tpe: ArrayType, elements: Vec<Scalar>) -> Self {
+    pub fn new(tpe: ArrayType, elements: impl IntoIterator<Item = impl Into<Scalar>>) -> Self {
```

Comment:
```suggestion
    #[cfg(any(test, feature = "test-ffi"))]
    pub fn new(tpe: ArrayType, elements: impl IntoIterator<Item = impl Into<Scalar>>) -> Self {
```

Let's keep this as private as possible

Reply:
test-ffi is only defined in the ffi crate, but this is in the kernel crate. Configuring this as test will also break `test-ffi` because the kernel code is no longer public to the ffi crate. 😔

Reply:
mmm, okay. maybe at least don't make it pub by default and only if `developer-visibility` is specified, which the ffi already asks for.
```suggestion
    #[cfg_attr(feature = "developer-visibility", visibility::make(pub))]
    fn new(tpe: ArrayType, elements: impl IntoIterator<Item = impl Into<Scalar>>) -> Self {
```